ansible/config_sonic_basedon_testbed.yml

```diff
@@ -75,10 +75,15 @@
       VM_topo: "{% if 'ptf' in topo %}False{% else %}True{% endif %}"
       remote_dut: "{{ ansible_ssh_host }}"
 
-  - name: gather testbed VM informations
+  - name: gather testbed VM informations from testbed_facts
     testbed_vm_info: base_vm={{ testbed_facts['vm_base'] }} topo={{ testbed_facts['topo'] }} vm_file={{ vm_file }}
     delegate_to: localhost
-    when: "VM_topo | bool"
+    when: "VM_topo | bool and testbed_name is defined"
+
+  - name: gather testbed VM informations from extra variables
+    testbed_vm_info: base_vm={{ vm_base }} topo={{ topo }} vm_file={{ vm_file }}
```

Comment:
The updated file `config_sonic_basedon_testbed.yml` is called by `testbed-cli.sh` in `generate_minigraph`, `deploy_minigraph` and `test_minigraph`. Let's take `generate_minigraph` as an example:

```bash
function generate_minigraph
{
  topology=$1
  inventory=$2
  passfile=$3
  shift
  shift
  shift

  echo "Generating minigraph '$topology'"

  read_file $topology

  ansible-playbook -i "$inventory" config_sonic_basedon_testbed.yml --vault-password-file="$passfile" -l "$duts" -e testbed_name="$topology" -e testbed_file=$tbfile -e vm_file=$vmfile -e local_minigraph=true $@

  echo Done
}
```
According to the `ansible-playbook` command line, variables `vm_base` and `topo` are not passed to the playbook. I suspect that in this playbook, variables `vm_base` and `topo` are undefined.